fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -276,7 +276,7 @@ protected String buildFromClause(String simpleName) {
         fromClause.append(" LR JOIN ");
         fromClause.append(simpleName);
         fromClause.append("_RESOURCES");
-        fromClause.append(" R ON R.LOGICAL_RESOURCE_ID=LR.LOGICAL_RESOURCE_ID AND R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID ");
+        fromClause.append(" R ON R.LOGICAL_RESOURCE_ID=LR.LOGICAL_RESOURCE_ID AND R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID AND R.IS_DELETED <> 'Y' ");
```

Comment:
Double checking the scenario.  We never search for Deleted resources? 
This is probably a great optimization.

Reply:
Is deleted has been always been checked, it was the first of the where clause before.